mllib/src/main/scala/org/apache/spark/ml/clustering/BisectingKMeans.scala

```diff
@@ -91,6 +91,9 @@ class BisectingKMeansModel private[ml] (
   extends Model[BisectingKMeansModel] with BisectingKMeansParams with MLWritable
   with HasTrainingSummary[BisectingKMeansSummary] {
 
+  @Since("3.0.0")
+  val numFeatures: Int = parentModel.clusterCenters.head.size
```

Comment:
This seems to be beyond the scope of this PR, but it looks okay. Maybe, `lazy val` is better?